spec/predicates/runtime-trace.md

```diff
@@ -0,0 +1,2677 @@
+
+## Prerequisites
+
+Understanding of monitoring tools and the larger in-toto attestation framework.
+
+## Model
+
+The structure of this predicate closely mirrors the structure of the SLSA Provenance attestation as they are complementary. `monitor` identifies the specific instance of a tool that’s observing the build, while `build` contains information that when put together uniquely identifies the exact instance of the job being observed. `invocation` details the configurations of `monitor` and the `monitorLog` contains the actual runtime trace information.
+
+This attestation can be used in conjunction with provenance attestations to leverage observability to attest to various SLSA requirements. The runtime trace can prove the build was invoked via a script, that the build was executed in a hermetic environment with no network access, and so on.
+
+## Schema
+
+The schema of the attestation type is the core part of the document. It defines the fields included in an instantiation of the attestation type, as well as rules to parse them.
+
+```json
+{
+  "_type": "https://in-toto.io/Statement/v0.1",
+  "predicateType": "https://in-toto.io/runtime-trace/v0.1.0",
+  "subject": [{ ... }],
+  "predicate": {
+    "monitor": {
+      "id": "<URI>"
+    },
+    “build”: {
+        “builderId”: “<URI>”,
+        “type”: “<URI>”,
+        “event”: “<STRING>”,
+    },
+    "invocation": {
+     "configSource": {
+        "uri": "<URI>",
+        "digest": { /* DigestSet */ },
+        "entryPoint": "<STRING>"
+      },
+      "tracePolicy": {}
+    },
+    "monitorLog": {
+      "process": [
+         { /* object */ }
+      ],
+      "fileAccess": [
+        { /* object */ }
+      ],
+    },
+    "metadata": {
+      "buildStartedOn": "<TIMESTAMP>",
+      "buildFinishedOn": "<TIMESTAMP>",
+    }
+  }
+}
+```
+
+### Parsing Rules
+
+This predicate follows the in-toto attestation parsing rules. Summary:
+
+-   Consumers MUST ignore unrecognized fields.
+-   The `predicateType` URI includes the major version number and will always change whenever there is a backwards incompatible change.
+-   Minor version changes are always backwards compatible and “monotonic.” Such changes do not update the `predicateType`.
+-   Producers MAY add extension fields using field names that are URIs.
+-   _optional_ fields MAY be unset or null, and should be treated equivalently. Both are equivalent to empty for object or array values.
+
+## Fields
+
+`monitor` _object_, _required_
+
+Identifies the specific monitor instance used to trace the runtime.
+
+`monitor.id` _string (URI)_, _required_
+
+URI indicating the monitor’s identity.
+
+`build` _object_, _required_
+
+Feedback required: Should this predicate be scoped to runtime traces of builds only or generalize to runtime traces of any activity?
```

Comment:
I think we need to either rename the attestation to mention build, or try to generalize the all the build-specific fields (most obvious example is the field called `build`). currently it doesn't make sense

Reply:
I've updated this to use `process` instead of `build`. WDYT?

cc @pxp928 @nadgowdas

Reply:
yes that looks good. Thanks @adityasaky